Description: update lis driver with libftdi1
 patch required to update libftdi to libftdi1
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=810373
Author: Daryl F <wyatt@prairieturtle.ca>
Origin: http://lists.omnipotent.net/pipermail/lcdproc/2016-January/014317.html
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -335,7 +335,7 @@
 				ac_cv_lis_pthread=no
 				AC_MSG_WARN([The lis driver needs pthread.h])
 			])
-			if test "$enable_libftdi" = yes ; then
+			if test "$enable_libftdi1" = yes ; then
 				if test "$enable_libusb" = yes; then
 					if test "$ac_cv_lis_pthread" = yes; then
 						DRIVERS="$DRIVERS lis${SO}"
@@ -347,7 +347,7 @@
 					AC_MSG_WARN([The lis driver needs the usb library])
 				fi
 			else
-				AC_MSG_WARN([The lis driver needs the ftdi library])
+				AC_MSG_WARN([The lis driver needs the ftdi1 library])
 			fi
 			;;
 		MD8800)
@@ -559,6 +559,7 @@
 AC_SUBST(LIBG15)
 AC_SUBST(LIBGLCD)
 AC_SUBST(LIBFTDI)
+AC_SUBST(LIBFTDI1)
 AC_SUBST(LIBXOSD)
 AC_SUBST(LIBPTHREAD_LIBS)
 AC_SUBST(LIBMX5000)
--- a/configure.ac
+++ b/configure.ac
@@ -323,6 +323,29 @@
 AC_SUBST(LIBFTDI_CFLAGS)
 
 dnl ######################################################################
+dnl libftdi1 support
+dnl ######################################################################
+AC_MSG_CHECKING([if libftdi1 support has been enabled]);
+AC_ARG_ENABLE(libftdi1,
+	[AS_HELP_STRING([--disable-libftdi1],[disable FTDI1 support using libftdi1])],
+	[ if test "$enableval" != "no"; then
+		enable_libftdi1=yes
+	fi ],
+	[ enable_libftdi1=yes ]
+)
+AC_MSG_RESULT($enable_libftdi1)
+
+if test "$enable_libftdi1" = "yes"; then
+	ifdef([PKG_CHECK_MODULES],
+	 	[PKG_CHECK_MODULES(LIBFTDI1, libftdi1 >= 1.0,
+			[AC_DEFINE(HAVE_LIBFTDI1, [1], [Define to 1 if you have libftdi1])],
+			[ enable_libftdi1=no ])],
+		[AC_MSG_WARN([pkg-config not (fully) installed; drivers requiring libftdi1 may not be built])])
+fi
+AC_SUBST(LIBFTDI1_LIBS)
+AC_SUBST(LIBFTDI1_CFLAGS)
+
+dnl ######################################################################
 dnl X11 library support
 dnl ######################################################################
 AC_MSG_CHECKING([if X11 support has been enabled]);
--- a/server/drivers/lis.c
+++ b/server/drivers/lis.c
@@ -42,7 +42,7 @@
 #include <pthread.h>
 
 #include <usb.h>
-#include <ftdi.h>
+#include <libftdi1/ftdi.h>
 
 #include "lcd.h"
 #include "lis.h"
--- a/server/drivers/Makefile.am
+++ b/server/drivers/Makefile.am
@@ -31,7 +31,7 @@
 hd44780_CFLAGS =     @LIBUSB_CFLAGS@ @LIBFTDI_CFLAGS@ $(AM_CFLAGS)
 i2500vfd_CFLAGS =    @LIBFTDI_CFLAGS@ $(AM_CFLAGS)
 IOWarrior_CFLAGS =   @LIBUSB_CFLAGS@ $(AM_CFLAGS)
-lis_CFLAGS =         @LIBUSB_CFLAGS@ @LIBFTDI_CFLAGS@ $(AM_CFLAGS)
+lis_CFLAGS =         @LIBUSB_CFLAGS@ @LIBFTDI1_CFLAGS@ $(AM_CFLAGS)
 mdm166a_CFLAGS =     @LIBHID_CFLAGS@ $(AM_CFLAGS)
 picolcd_CFLAGS =     @LIBUSB_CFLAGS@ @LIBUSB_1_0_CFLAGS@ $(AM_CFLAGS)
 shuttleVFD_CFLAGS =  @LIBUSB_CFLAGS@ $(AM_CFLAGS)
@@ -58,7 +58,7 @@
 lb216_LDADD =        libLCD.a
 lcterm_LDADD =       libLCD.a libbignum.a
 lirc_LDADD =         @LIBLIRC_CLIENT@
-lis_LDADD =          libLCD.a @LIBFTDI_LIBS@ @LIBUSB_LIBS@ @LIBPTHREAD_LIBS@ libbignum.a
+lis_LDADD =          libLCD.a @LIBFTDI1_LIBS@ @LIBUSB_LIBS@ @LIBPTHREAD_LIBS@ libbignum.a
 mdm166a_LDADD =      @LIBHID_LIBS@ libLCD.a
 mtc_s16209x_LDADD =  libLCD.a
 MtxOrb_LDADD =       libLCD.a libbignum.a
